### PR TITLE
feat: remove JSS manager dependency from React base components

### LINKED ISSUE #141: fast-components-react-base should not depends on fast-jss-manager-react
fast-components-react-base should be an agnostic on JSS usage, the only thing it defined is to have a managedClasses prop which follows IButtonClassNameContract  contract. In other word, the user of base components can use them w/ or w/o JSS (fast-jss-manager-react). @nicholasrice @awentzel.

https://github.com/Microsoft/fast-dna/blob/f666c238ce6d952896a6543f8ed60830a908b1de/packages/fast-components-react-base/src/button/Button.tsx#L5

### PR DESCRIPTION
**Pull request by nicholasrice**
closes #141

### CODE REVIEW

File: packages/fast-components-react-base/src/managed-classes.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+ export type ClassNames<T> = {
+    /**
```
Comment (chrisdholt) [inline packages/fast-components-react-base/src/managed-classes.ts]: Should this comment be above our export statement?

Comment (nicholasrice) [reply packages/fast-components-react-base/src/managed-classes.ts]: sure, moved

**Comment by nicholasrice** [comment]
@K05-MSFT I had that thought also - the reason I chose not to was the contracts really shouldn't care where they're implemented, so to me it seemed out of place to come from that package. There really doesn't seem to be any good central location for that interface to live - components and the manager both need to know about it, but neither should take a dependency on the other.

I also wanted to keep the JSS manager as loosely coupled to other packages as possible. It won't add bloat since the interfaces are compiled out, the contracts didn't seem like a natural dependency for the manager.

With this implementation, we still get type safety from TypeScript's structural type system without having to add dependencies. It is definetely duplication which I generally try to avoid, but it seemed preferable to creating what I see as un-intuitive relationships.

**Comment by K05-MSFT** [comment]
@nicholasrice The whole ```fast-components-class-name-contracts``` will contains nothing in the final JS output, but gets used to contain type def info only. The dependency on it won't introduce any production bits, so, IMO, it is better than un-intuitive and duplication type def in term of "More readable" and easy maintenance. In term of type safety, we do lose some type safety due to the duplication def, since ```fast-jss-manager-react``` and ````fast-components-react-base``` are independent components, it won't get caught during the build time of each, for example, the contract gets updated only one side but not the other.

**Comment by nicholasrice** [comment]
@K05-MSFT We don't get type safety *between* the two but we do get type safety where the two components are used together - changing the interface in either will prevent typescript from compiling `fast-components-react-msft` (and thus fail the build) because the two now do not have identical type structures.

I see your point though, and sharing the interface is more safe so I'll change to your suggestion. Thanks for the feedback!

**Comment by K05-MSFT** [comment]
And, ```fast-components-class-name-contracts``` is more like a ```devDependencies``` instead of ```dependencies``` given it is only used during the development, build time only particularly.

**Comment by K05-MSFT** [comment]
It seems reasonable to ask the user to have it in its ```devDependencies```, essentially, the package is only used for development in any cases. Keep it in ```dependencies``` is just for convenient of the user, typescript user particularly, but not production 'dependencies' indeed. Think about there is case package's type def are shipped in separated package alone, like ```@types/tspackage```, which should get consumed as ```devDependencies```. Thought?

**Comment by nicholasrice** [comment]
I agree in principle, but to me it is non-starter to have a TS app fail to compile because we're shipping `d.ts` that do not have their dependencies enumerated. That is not a good first-experience. [www.typescriptlang.org](http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) actually addressed this issue in their documentation and their recommendation is to use `dependencies`.

**Comment by K05-MSFT** [comment]
Give it is suggested by TS lang, I am fine with that.

**Comment by nicholasrice** [comment]
Great, thanks Knight!

**Pull request merged**